chore: improve esbuild tests to download snapshots automatically

sapphi-red opened the pull request:
Improved the snap-diff script to download snapshots automatically. Also specified the version of esbuild so that we can bump it automatically later on.

sapphi-red commented:
* **#7379** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7379?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7378** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7378?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7378?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7377** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7377?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7376** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7376?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=7378).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 reviewed (approved):
What part makes the download `automatically`? Is it part of `just setup`?

sapphi-red commented:
It is downloaded when you run this script (e.g. `pnpm esbuild-snap-diff` in `scripts`)

sapphi-red committed:
chore: improve esbuild tests to download snapshots automatically (#7378)

Improved the snap-diff script to download snapshots automatically. Also specified the version of esbuild so that we can bump it automatically later on.
